### PR TITLE
chore: generalize watch mode in packages and document process in contributing guidelines

### PR DESCRIPTION
**Pull request by dhayab**
**Summary**

This PR adds a `watch` script on almose all relevant packages in the monorepo and allows them to be run in parallel using lerna. It also updates the CONTRIBUTING.md guidelines with a description of the process.

> [!NOTE]
> **algoliasearch-helper** is omitted from this PR as it would a more significant refactoring of its build script to make it work properly in watch mode

### CODE REVIEW

File: packages/vue-instantsearch/rollup.config.js
Comment (dhayab) [inline packages/vue-instantsearch/rollup.config.js:1]: While not necessary, this helps reduce the build time in watch mode as we only need the vue2/es build.

**Review by Haroenv — COMMENTED** [review]
maybe the watch script should be called `watch:es` `watch:build-minimal` or similar, just to make it clear that in some edge cases it doesn't reflect all changes (like when you're testing out cjs). Overall looks ok though

File: packages/instantsearch-ui-components/package.json [hunk 1]
```diff
@@ -39,10 +39,12 @@
...
     "clean": "rm -rf dist",
     "build": "yarn build:cjs && yarn build:es && yarn build:types",
-    "build:es": "BABEL_ENV=es babel src --root-mode upward --extensions '.js,.ts,.tsx' --out-dir dist/es --ignore '**/__tests__/**/*','**/__mocks__/**/*' --quiet",
+    "build:es:base": "BABEL_ENV=es babel src --root-mode upward --extensions '.js,.ts,.tsx' --out-dir dist/es --ignore '**/__tests__/**/*','**/__mocks__/**/*'",
+    "build:es": "yarn build:es:base --quiet",
     "build:cjs": "BABEL_ENV=cjs babel src --root-mode upward --extensions '.js,.ts,.tsx' --out-dir dist/cjs --ignore '**/__tests__/**/*','**/__mocks__/**/*' --quiet && ../../scripts/prepare-cjs.sh",
     "build:types": "tsc -p ./tsconfig.declaration.json --outDir ./dist/es",
-    "version": "./scripts/version.cjs"
+    "version": "./scripts/version.cjs",
+    "watch": "yarn --silent build:es:base --watch"
```
[the remark this answers is not in the document]
Comment (Haroenv) [reply packages/instantsearch-ui-components/package.json:47]: another aspect of this is not building `types`, that means that when you enable this `watch` mode, you'll often have TS errors as the types will be removed

[the remark this answers is not in the document]
Comment (dhayab) [reply packages/instantsearch-ui-components/package.json:47]: The aim of this command is just to ease the DX process when contributing, as examples is importing from es modules. I can rename this command to `watch:es` to make it more explicit that it only rebuilds this particular build type.

[the remark this answers is not in the document]
Comment (Haroenv) [reply packages/instantsearch-ui-components/package.json:47]: lgtm if it's called `watch:es` globally then :)

**Review by Haroenv — APPROVED** [review]

**Pull request merged**